tensorboard/components/experimental/plugin_util/plugin_api_host_test.ts

```diff
@@ -0,0 +1,267 @@
+    await TestBed.configureTestingModule({
+      providers: [
+        provideMockStore(),
+        {provide: Ipc, useClass: NoopIpc},
+        PluginRunsApiHostImpl,
+        PluginCoreApiHostImpl,
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+    store.overrideSelector(getExperimentIdsFromRoute, ['e1']);
+    store.overrideSelector(getRuns, []);
+    selectSpy = spyOn(store, 'select').and.callThrough();
+
+    const ipc = TestBed.inject(Ipc);
+    broadcastSpy = spyOn(ipc, 'broadcast');
+    listenSpy = spyOn(ipc, 'listen');
+    coreApi = TestBed.inject(PluginCoreApiHostImpl);
+    runApi = TestBed.inject(PluginRunsApiHostImpl);
+  });
+
+  describe('runs apis', () => {
+    describe('#experimental.RunsChanged', () => {
+      it('broadcasts runs when runs change', () => {
+        runApi.init();
+        // initial `runs` are flushed.
+        expect(broadcastSpy).toHaveBeenCalledTimes(1);
+        expect(broadcastSpy).toHaveBeenCalledWith(MessageId.RUNS_CHANGED, []);
+
+        store.overrideSelector(getRuns, [
+          buildRun({
+            id: '1',
+            name: 'hello',
+          }),
+          buildRun({
+            id: '2',
+            name: 'world',
+          }),
+        ]);
+        store.refreshState();
+
+        expect(broadcastSpy).toHaveBeenCalledTimes(2);
+        expect(broadcastSpy).toHaveBeenCalledWith(MessageId.RUNS_CHANGED, [
+          'hello',
+          'world',
+        ]);
+      });
+
+      it('broadcasts runs combining all runs in current experiments', () => {
+        store.overrideSelector(getExperimentIdsFromRoute, ['exp1', 'exp2']);
+        const runsForExp = new Map([
+          [
+            'exp1',
+            [
+              buildRun({
+                id: '1',
+                name: 'hello',
+              }),
+              buildRun({
+                id: '2',
+                name: 'world',
+              }),
+            ],
+          ],
+
+          [
+            'exp2',
+            [
+              buildRun({
+                id: '3',
+                name: 'hello',
+              }),
+              buildRun({
+                id: '2',
+                name: 'world',
+              }),
+              buildRun({
+                id: '1',
+                name: 'TensorBoard',
+              }),
+            ],
+          ],
+        ]);
+        selectSpy
+          .withArgs(getRuns, jasmine.any(Object))
+          .and.callFake(
+            (
+              selector: typeof getRuns,
+              {experimentId}: Parameters<typeof getRuns>[1]
+            ) => {
+              return of(runsForExp.get(experimentId));
+            }
+          );
+        runApi.init();
+        expect(broadcastSpy).toHaveBeenCalledWith(MessageId.RUNS_CHANGED, [
+          'hello',
+          'world',
+          'TensorBoard',
+        ]);
+      });
+
+      it('broadcasts only when runs changes', () => {
```

Comment:
Ah, I see that the existing behavior was only checking run names, since the Polymer `runsStore` did not know about `ids`.  Now that we are hooking into the Ngrx store, I think it makes sense to use `id`s now that we can, as that is closer to the intent of the API.  e.g. if someone deletes a run from filesystem and creates a new one with the same name, an author probably wants to be notified.

Wdyt about changing `setOnRunsChanged` to emit whenever run `id`s change, but preserve `getRuns()` behavior of returning run names, in this PR?